singleflight/example_test.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2018 The Go Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package singleflight_test
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"time"
+
+	"github.com/golang/groupcache/singleflight"
+)
+
+var counter int32
+var group singleflight.Group
+
+func Example() {
```

Comment:
Should this be a testing case, like TestSingleFlightExample, and use `testing.T` for checking?